smslibrary/src/main/java/com/eis/communication/MessageHandler.java

```diff
@@ -4,23 +4,26 @@
  * Class used to switch from actual communication data to Message and back
  * Should use a <a href='https://en.wikipedia.org/wiki/Singleton_pattern'>Singleton pattern</a>
  *
+ * @param <D> The data type used by the channel
+ * @param <P> The peer type used by the channel
  * @param <M> The type of message to parse and output
  */
-public interface MessageHandler<M extends Message> {
+public interface MessageHandler<D,P,M extends Message> {
 
     /**
-     * Interprets a string arrived via the communication channel and parses it to a message
+     * Interprets data arrived via the communication channel and parses it to a message
      *
-     * @param data data from the communication channel
-     * @return the message if the string has been parsed correctly, null otherwise
+     * @param peerData from the communication channel
+     * @param data from the communication channel
+     * @return the message if the data has been parsed correctly, null otherwise
      */
-    M parseMessage(String data, String peerData);
+    M parseMessage(P peerData, D data);
```

Comment:
Why did u decide to use "_peerData_" instead of "_peer_" ?

Reply:
Because what's represented it's not actually a peer class, it's just its data. Like if we were on IP: `String peerData = 192.168.1.1` while `IPAddress peer = new Peer('192.168.1.1')`, they represent two different things